addons/api/mirage/factories/target.js

```diff
@@ -23,22 +23,28 @@ export default factory.extend({
       'remove-credential-sources',
     ],
 
-  id: () => generateId('t_'),
-
   /**
    * -1 means "unlimited" and we want to generate these on occasion.
    */
   session_connection_limit: () =>
     faker.helpers.arrayElement([-1, faker.datatype.number()]),
   worker_filter: (i) => (i % 2 === 0 ? faker.random.words() : null),
+  egress_worker_filter: (i) => (i % 2 !== 0 ? faker.random.words() : null),
+  ingress_worker_filter: (i) => (i % 2 !== 0 ? faker.random.words() : null),
   type: (i) => types[i % types.length],
   /**
    * Generates attributes fields by type.
    */
   afterCreate(target) {
+    const id =
```

Comment:
Is there a reason the id is being created in this `afterCreate` instead of as part of this `target` class as it was previously? 

Regarding the error you're seeing, I would guess it's because of this since it doesn't have an id until after being created and the `host-set` has an association with `targets` in the models and isn't getting the right id because it doesn't have one yet.

Reply:
I moved it to the `afterCreate` to give a more accurate id based on the target type. However, I can see that not being a good trade off for the test setup changes I had to make to accommodate this change. Happy to change it back if you feel I should.

Reply:
You are right. I am going to try what I am thinking and push it up shortly. Thanks for the direction!